1. Listázzuk a <span style="color: #800000;">`kolcsonzes`</span> tábla következő oszlopait: ID, ki\_datum, vissza\_datum! 

  

a.  Az ID oszlop neve legyen Azonosító

b.  A ki\_datum oszlop neve legyen Kölcsönzés\_dátuma

c.  A vissza\_datum oszlop neve legyen Visszahozatal\_dátuma

In [2]:
SELECT ID AS 'Azonosító', --kölcsönző azonosítója
        ki_datum AS 'Kölcsönzés dátuma',
        vissza_datum AS 'Visszahozatal dátuma'
FROM kolcsonzesek 

(5 rows affected)

Total execution time: 00:00:00.005

Azonosító Kölcsönzés dátuma Visszahozatal dátuma 10 2003-12-11 NULL 11 2003-12-11 NULL 12 2003-12-11 NULL 13 2006-10-01 2006-10-10 14 2006-10-02 2006-10-09

2. Melyek azok a kölcsönzések, amelyek 2005.01.01 és 2007.01.01 között történtek?

In [4]:
SELECT *
FROM kolcsonzesek
WHERE ki_datum BETWEEN '2005.01.01' AND '2007.01.01' --végpontok is benne vannak 
-- WHERE ki_datum >= '2005.01.01' AND 

(2 rows affected)

Total execution time: 00:00:00.004

ID ki mit ki_datum vissza_datum 13 1 3 2006-10-01 2006-10-10 14 2 3 2006-10-02 2006-10-09

3. Melyek azok a tagok, akiknek kedvenc stílusa a játék, és nevük a-ra végződik?

  

a.  Csak a tagok neve jelenjen meg, az oszlop neve Név legyen

In [6]:
SELECT nev AS 'név'
FROM tagok 
WHERE nev LIKE '%a' AND 
    kedvencstilus='játék'

(2 rows affected)

Total execution time: 00:00:00.004

név Bodó Kinga Garay Magdolna

4. Listázzuk egy oszlopban a tagok nevét, címét és kedvenc stílusát összefűzve! 

  

a.  A nevet, címet és a stílust kötőjellel válasszuk el

b.  A kötőjelek előtt és után legyen 1-1 szóköz

c.  Az oszlop neve legyen TAG

d.  Kinga keresztnevű tagok ne legyenek benne a listában

In [ ]:
valami megint

5. Mennyi lenne a dvd-k nettó ára, ha 20%-kal csökkenne? 

  

a.  Csak a dvd-k címét és a csökkentett árat jelenítsük meg

b.  A csökkentett árat nevezzük el diszkont\_ar-nak

In [14]:
SELECT cim AS 'cím',
        nettoar * 0.8 AS 'diszkont ár'
FROM dvd

(318 rows affected)

Total execution time: 00:00:00.008

cím diszkont ár NULL 3087,2000000000003 A II. világháború története (1999) 3395,2000000000003 3D Dinosaur - adventure 4032,8 3D virtuális origami (2000) 4284 Az 1848-1849-iki szabadságharc története (2002) 4560 Ablak zsiráf (1999) 2832 Ady Endre összes művei (1999) 3392,8 Afrika 4254,400000000001 Age of Empires (2000) 3585,6000000000004 Airport Inc. 3448,8 American Rodeo : BullRider 2000 (2001) 4472,8 Angol : 8-10 éveseknek (2002) 3882,4 Angol és német nyelvtani gyakorlatok (2002) 3924,8 Angol szótanító CD-ROM 3559,2000000000003 Angol szótanító CD-ROM 3555,2000000000003 Angol szótár (1996) 2807,2000000000003 Angol-magyar, Magyar-angol hangosszótár 4120,8 Angol-magyar nagyszótár (Halász Előd) 4510,400000000001 Az áramvonalas fakanál 4564,8 Arany János: A bajusz (2002) 2712,8 Az Árpád-ház története 4017,6000000000004 Art Attack (1998) 3131,2000000000003 Art puzzle (1998) 4628,8 Az ásványok világa 4499,2 Ausztrália 4260,8 Auto 2000 - Kerékvilág 3570,4 Ázsia (2000) 2498,4 Biblia (1995) 3552 Biológia '97 4434,400000000001 Biológia (2000) 3055,2000000000003 Biológia (2001) 3033,6000000000004 Biológia : 12-14 éveseknek (2003) 2785,6000000000004 Blinky Bill : A fantasztikus léggömbkaland 3260,8 Broken Sword II. (cop. 1997) 3601,6000000000004 Budapest kalauz 3243,2000000000003 Budapest Museum 4233,6 Buggy 4664 Byte Magyarország (2002) 2621,6000000000004 Caesar III. 3308,8 Capitalism II. (2001) 4849,6 Capitalism Plus 4024 Castle Explorer 2965,6000000000004 CD Panoráma 1-4. (1998) 4337,6 Computer iskola (1998) 4208 Csizmás kandúr (1998) 4745,6 Csodálatos univerzum (1997) 3269,6000000000004 Dél-Amerika (1999) 3199,2000000000003 Dementia 1-5. (1997) 4735,2 Deutsch: Anfanger (kezdő) + Telepítő 4506,400000000001 Deutsch: Fortgeschrittene (haladó) + Installation 4627,2 Deutsch: Mittelstufe (középhaladó)+ Telepítő 3501,6000000000004 Diablo 4775,2 Diablo II. (2000) 3920 Dicső napok és hétköznapok 1848 - Digitális forrásgyűjtemény 1. 4488,8 Dreams to Reality 1-2. (1997) 3759,2000000000003 Duke Nukem 3D 4379,2 Dungeon Keeper 2. (1999) 2695,2000000000003 E-Games : Collector's Edition (2001) 4466,400000000001 Éléstár (1998) 2606,4 Élet a polgárosodó Magyarországon (2001) 4457,6 Életképek 1848 (1998) 3368,8 Az ember tragédiája 3436 Encarta World Atlas '99 1-2. (1998) 4267,2 Enemy Engaged RAH-66 4173,6 English 1-2. + Telepítő 4172 English: Intermediate (haladó) 2901,6000000000004 English by Stealth lásd: Lopva angolul 3582,4 Erdély (1996) 3999,2000000000003 Espanol: Intermedio + Telepítő 3131,2000000000003 Espanol: Principiante + Telepítő 4705,6 Észak-Amerika 3195,2000000000003 Európa 4216 The F.A. Premier League Stars 2001 3096,8 Felvételi példatár : Angol 3100,8 Felvételi példatár : Biológia, földrajz 2572 Felvételi példatár : Kémia, fizika 4680,8 Felvételi példatár : Magyar nyelv és irodalom 4764 Felvételi példatár : Matematika 4007,2000000000003 Felvételi példatár : Német 2481,6000000000004 FIFA, 2000 (1999) 4047,2000000000003 Fighting Force (1998) 3840 Final Liberation (1997) 3705,6000000000004 Fitness mesterfokon / Béres Alexandra (2001) 3685,6000000000004 Fizika : 12-14 éveseknek (2002) 2987,2000000000003 Fizika felkészítő az érettségi és felvételi vizsgára 3848 Fizika világa (2001) 4610,400000000001 A Föld élővilága 4720 Földrajz: 10-14 éveseknek (2002) 3144 Földrajz: érettségi előkészítő (2002) 2538,4 Francais 1-2. + Telepítő 3230,4 Francais: Débutan (kezdő) + Telepítő 3016 Francais: Moyen + Telepítő 4495,2 Francia-magyar szótár 2852,8 Funschool in Dreamland (1995) 3812,8 Geolexikon 4.2 : földtani adatbázis (2002) 3681,6000000000004 Gépírás 2000 3148,8 Gondolattár: aforizmák, szállóigék 4276,8 GTA 2 4004 Hadiflották I. (2002) 4099,2 Hadtudományi lexikon (1998) 2812 Hajdúböszörmény 3024,8 Halak, kétéltűek, hüllők 3322,4 Half-Life (1996) 4324 Half-Life Generation: Game of the Year Edition (2002) 2964 Half-Life Generation: Counter Strike (2002) 4595,2 Half-Life Generation: Blue Shift (2002) 4582,400000000001 Harry Potter és a bölcsek köve (2001

6. Melyek azok a kölcsönzések, ahol még nem hozták vissza a dvd-t?

In [12]:
SELECT *
FROM kolcsonzesek
WHERE vissza_datum  

(0 rows affected)

Total execution time: 00:00:00.046

ID ki mit ki_datum vissza_datum

7. Listázzuk azon diákok nevét és születési idejét, akik 1984-ben vagy 1985-ben születtek!  

  

a.  A születési dátum csak az évet, hónapot és a napot tartalmazza! 

b.  A születési dátum oszlopát nevezzük el 'Születési idő'-nek!

In [23]:
SELECT nev AS 'Név',
        CAST(Szulido AS date) AS 'Születési idő'

        --left(cast(szulido as nvarchar(25)), 11)

        --cast(YEAR(szulido) as nvarchar(4)) + '-' +
        --cast(month(szulido) as nvarchar(2)) +  '-' +
        --cast(DAY(szulido) as nvarchar(2)) 

FROM diak
WHERE YEAR(Szulido) = '1984' OR YEAR(Szulido) = '1985'
--where year(szulido) IN (1984,1985)

(3 rows affected)

Total execution time: 00:00:00.005

Név (No column name) Kovács Imre Mar 2 1984 Kapos Petra Dec 23 1985 Koppány Olga Feb 28 1984

8. Készítsünk lekérdezést, amely a tanulók nevét és az ebből képzett felhasználói nevét tartalmazza. 

  

a.  A felhasználói név álljon a név első kettő, illetve utolsó kettő betűjének összefűzéséből!

b.  A listát szűrjük azon tanulókra, akik teljes neve - szóközzel együtt - legalább 10 betűs!

In [25]:
SELECT LEFT(nev, 2) + RIGHT(nev, 2)
FROM diak
WHERE LEN(nev)>=10

(8 rows affected)

Total execution time: 00:00:00.005

(No column name) Kore Hoil Kara Csna Nyás Orma Koga Koia

9. Készítsünk listát a diákok adatairól, ahol a diákok neve úgy jelenik meg, hogy először a keresztnév, majd a vezetéknév, közöttük szóközzel!  

a.  Ezen felül a születési idő három külön oszlopban jelenjen meg (év, hónap, nap). 

b.  Az oszlopokat nevezzük el értelemszerűen!

10. Készítsünk listát a munkák azonosítójáról és megnevezéséről! Egy új oszlopban azt is jelenítsük meg, hogy melyik munka hetente mennyit fizet! 

a.  Az oszlop neve legyen Heti bér, az értéket kerekítsük 1000 forintra! 

b.  A listát szűrjük azon rekordokra, ahol a kerekített heti bér 10000 Ft felett van!

In [35]:
SELECT munkaId, 
       allas,
       ROUND(oradij * oraszam * 5, -3)  AS 'Heti bér'
FROM munka
WHERE  ROUND(oradij * oraszam * 5, -3) > 10000

(7 rows affected)

Total execution time: 00:00:00.011

munkaId allas Heti bér 5 kézbesítő 14000 7 eladó 12000 16 pincér 14000 17 pincér 14000 20 takarító 11000 21 takarító 11000 41 eladó 12000

11. A diákok számára differenciált béremelést terveznek: a középiskolások esetében 33%, egyéb esetben 17% mértékben. Készítsünk listát, amely tartalmazza a munkák azonosítóját, az állás nevét, az eredeti óradíjat és a tervezett emelt óradíjat.

  

a.  Az oszlopoknak adjuk nevet értelemszerűen!

b.  Az emelt óradíj összegét kerekítsük egészre! (Az esetlegesen megjelenő 0 tizedesjegyekkel ne foglalkozzunk)

12. A diákok számára próbaidőt írnak elő, amely a munka kezdetétől számítva 3 hónapig tart.    Jelenítsük meg a munka tábla adatait egy új oszloppal kiegészítve, amelyik a próbaidő    végének dátumát mutatja!  

  

a.  Az oszlop neve Próbaidő vége legyen! 

b.  A lista ne tartalmazza azokat a munkákat, ahol a diákok azonosítója nincs megadva!

13. Készítsünk listát a munkák azonosítóiról és a megnevezésükről! A megnevezés allas-óraszám formában jelenjen meg, pl: eladó-4. 

  

a.  Jelenítsük meg a dátumot is, de csak a évet és a hónapot, pl: 2003-07! 

b.  Ennek az oszlopnak 'Kezdés hónapja' legyen a neve! A listát szűrjük a nyári hónapokra!

14. Készítsünk listát a diákok adatairól, amely a Nev oszlop helyén a diák monogramját jeleníti meg (keresztnév, illetve vezetéknév első betűje összefűzve). 

a.  A listát szűrjük 1983.01.01 és 1987.06.01 között született diákokra!

15. A diákok a munkakezdés évének utolsó napján bónuszt kapnak, amennyiben az adott évben legalább 6 hónapot dolgoznak. 

  

a.  Jelenítsük meg, hogy az egyes munkák adatait, egy új oszlopban az éves bónusz napját is 'Bónusz dátum' néven! 

b.  A lista csak a bónuszra jogosultak adatait tartalmazza!